src/utils/flattenChildren/flattenChildren.ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// Code from https://github.com/gregberge/react-flatten-children
+import * as React from "react";
+
+type ReactChildArray = ReturnType<typeof React.Children.toArray>;
+
+export function flattenChildren(children: React.ReactNode): ReactChildArray {
+  const childrenArray = React.Children.toArray(children);
+
+  return childrenArray.reduce((flatChildren: ReactChildArray, child) => {
+    if ((child as React.ReactElement<any>).type === React.Fragment) {
+      return flatChildren.concat(
+        flattenChildren((child as React.ReactElement<any>).props.children)
+      );
+    }
+
+    flatChildren.push(child);
+    return flatChildren;
+  }, []);
+}
```

Comment:
What will you say about following change? We'll get rid of mutability then and I think creating another array in `reduce` is pretty much last resoty
```suggestion
export function flattenChildren(children: React.ReactNode): ReactChildArray {
    const childrenArray = React.Children.toArray(children);

    return childrenArray.flatMap((child) => {
      if ((child as React.ReactElement<unknown>).type === React.Fragment) {
        return flattenChildren((child as React.ReactElement<any>).props.children);
      }

      return child;
    }, []);
  }
```

Reply:
I didn't wanted to modify this module because it's just pure copy from https://github.com/gregberge/react-flatten-children :)

Reply:
I understand, but if we'll realign our linters with overwatch, this code will not pass them. So I think it's better to do it in advance. WDYT?

Reply:
Sure, good idea, after all I can modify comment from "Code from" to "Based on" 🙈